src/server.ts

```diff
@@ -2,12 +2,91 @@ import dotenv from 'dotenv'
 dotenv.config()
 
 import express from 'express'
+import { connectDatabase, getUserCollection } from './app/utils/database'
+import { nanoid } from 'nanoid'
+
+if (!process.env.MONGODB_URI) {
+  throw new Error("Couldn't connect to the database")
+}
 
 const app = express()
 const port = process.env.PORT || 3001
 
 app.use(express.json())
 
+app.post('/register/', async (req, res) => {
+  const newUser = req.body
+  const existingUser = await getUserCollection().findOne({
+    username: newUser.username,
+  })
+  if (existingUser) {
+    res
+      .status(409)
+      .send(`Account with username ${newUser.username} already exists!`)
+  } else {
+    await getUserCollection().insertOne(newUser)
+    res.status(201).send(`Account was created`)
+  }
+})
+
+app.post('/login', async (req, res) => {
+  const user = req.body
+  const existingUser = await getUserCollection().findOne(
+    { username: user.username },
+    { projection: { _id: 0, username: 1, password: 1 } }
+  )
+  if (existingUser && existingUser.password === user.password) {
+    res.status(200).send('Login successful')
+  } else {
+    res.status(403).send('Incorrect username or passwort')
+  }
+})
+
+app.patch('/:username/concerts/add', async (req, res) => {
+  const username = req.params.username
+  const concert = { id: nanoid(), ...req.body }
+  const insertedConcert = await getUserCollection().updateOne(
+    { username },
+    {
+      $push: {
+        concerts: {
+          id: concert.id,
+          mainAct: concert.mainAct,
+          support: concert.support,
+          concertDate: concert.concertDate,
+          location: concert.location,
+          numberOfTickets: concert.numberOfTickets,
```

Comment:
the concert object is exactly structured as the required structure in the db. then you can just push the concert object 
`$push: {
  concerts: concert
}`